### ISSUE TITLE
AlexNet problem Batch Normalization caffe-0.16

### ISSUE THREAD

**Issue by Wronskia** [tidsp/caffe-jacinto#7]
Hello,

I am trying to finetune AlexNet with sparisification. I get the following error in caffe-jacinto (caffe-0.16) which i don't get when running the same process in caffe  : 

I0929 15:28:49.480206 19322 net.cpp:184] Created Layer fc6/bn (22)
I0929 15:28:49.480216 19322 net.cpp:561] fc6/bn <- fc6
I0929 15:28:49.480231 19322 net.cpp:530] fc6/bn -> fc6/bn
F0929 15:28:49.480550 19322 blob.hpp:252] Check failed: axis_index < num_axes() (2 vs. 2) axis 2 out of range for 2-D Blob with shape 100 4096 (409600)
*** Check failure stack trace: ***
    @     0x7f8d746ac5cd  google::LogMessage::Fail()
    @     0x7f8d746ae433  google::LogMessage::SendToLog()
    @     0x7f8d746ac15b  google::LogMessage::Flush()
    @     0x7f8d746aee1e  google::LogMessageFatal::~LogMessageFatal()
    @           0x417f58  caffe::Blob::CanonicalAxisIndex()
    @     0x7f8d75519cce  caffe::BatchNormLayer<>::LayerSetUp()
    @     0x7f8d756e2f29  caffe::Net::Init()
    @     0x7f8d756e492e  caffe::Net::Net()
    @     0x7f8d752b3210  caffe::Solver::InitTrainNet()
    @     0x7f8d752b374d  caffe::Solver::Init()
    @     0x7f8d752b3c22  caffe::Solver::Solver()
    @     0x7f8d752c0126  caffe::Creator_SGDSolver()
    @           0x41bce6  caffe::SolverRegistry::CreateSolver()
    @           0x4133e2  train()
    @           0x40deb8  main
    @     0x7f8d72de0830  __libc_start_main
    @           0x40eaf9  _start
    @              (nil)  (unknown)
root@6409698f2bc4:/workspace/imagenet-finetuning/Alexnet# 

 Best

**Comment by mathmanu** [comment]
Can you check if you get the same error when you train with https://github.com/NVIDIA/caffe branch caffe-0.16

**Comment by Wronskia** [comment]
You are right the same issue remains in nvidia fork of caffe, I will write an issue there.
